### PR TITLE
add copy action to saved question

### LINKED ISSUE #4483: Allow copying an existing question
I know it's possible to do this by opening a question, then changing a parameter, then saving as a new question. 

While that functionality is perfect and completely satisfactory, it was very confusing for me to find it, so it might be worth duplicating the functionality under "edit" an existing question. So maybe have a duplicate / save as copy icon at the top (next to the save changes)

⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### PR DESCRIPTION
**Pull request by daltojohnso**
Resolves #4483 

**Description**
Adds a copy action to the "pencil" dropdown on a saved question. Adding a bit of logic to `SaveQuestionModal`. Not sure if I should `omit` any of the `card` properties before I POST it to `/api/card`, but I went ahead and removed `id` from the object.

**Verification**
Current appearance
Screen Shot 2021-02-12 at 4 24 00 PM
Screen Shot 2021-02-12 at 4 24 05 PM

### CODE REVIEW

**Comment by daltojohnso** [comment]
@mazameli & @kdoh -- anything you want changed about this? action placement, text, modal text, etc

File: frontend/src/metabase/containers/SaveQuestionModal.jsx [hunk 1]
```diff
@@ -90,6 +92,8 @@ export default class SaveQuestionModal extends Component {
 
     const title = this.props.multiStep
       ? t`First, save your question`
+      : this.props.clone
+      ? t`Copy question`
```
Comment (mazameli) [inline frontend/src/metabase/containers/SaveQuestionModal.jsx:96]:
```suggestion
      ? t`Duplicate this question`
```

File: frontend/src/metabase/query_builder/components/view/QuestionEntityMenu.jsx [hunk 2]
```diff
@@ -41,6 +41,11 @@ export default function QuestionEntityMenu({
           title: `Archive`,
           action: () => onOpenModal("archive"),
         },
+        canWrite && {
+          icon: "clone",
+          title: t`Copy this question`,
```
Comment (mazameli) [inline frontend/src/metabase/query_builder/components/view/QuestionEntityMenu.jsx:46]:
```suggestion
          title: t`Duplicate this question`,
```

**Review by mazameli — CHANGES_REQUESTED** [review]
Nice. Just a couple small copy and ordering change requests:

- On dashboards we call this action "duplicate" so we should use the same language here.
- I think this action should be second to the bottom, right above Archive.

**Comment by mazameli** [comment]
Looking good now. One last hopefully simple detail I noticed: when duplicating dashboards, in the name field in the modal we add ` - Duplicate` to the end of the string, and I think we should do that here as well. Here's what it looks like for dashboards:

image

**Comment by daltojohnso** [comment]
@mazameli - That dashboard duplication modal is reusable so I was able to make that change and also remove my changes to the `SaveQuestionModal`. Glad you noticed that!

File: frontend/src/metabase/query_builder/components/QueryModals.jsx [hunk 3]
```diff
@@ -181,6 +182,24 @@ export default class QueryModals extends React.Component {
...
+      <Modal onClose={onCloseModal}>
+        <EntityCopyModal
+          entityType="questions"
+          entityObject={this.props.card}
+          copy={async formValues => {
+            const object = await this.props.onCreate({
+              ...this.props.card,
+              ...formValues,
+              description: formValues.description || null,
+              id: undefined,
```
Comment (paulrosenzweig) [inline frontend/src/metabase/query_builder/components/QueryModals.jsx:195]: I didn't test, but I don't think you need this line. `EntityCopyModal` passes `dissoc(entityObject, "id")`

**Review by paulrosenzweig — APPROVED** [review]

**Review by mazameli — APPROVED** [review]

File: frontend/src/metabase/query_builder/components/QueryModals.jsx (hunk 3 above)
Comment (daltojohnso) [reply frontend/src/metabase/query_builder/components/QueryModals.jsx]: You're right -- removed `id: undefined`

**Pull request merged**